sh/drive-utils.sh

```diff
@@ -0,0 +1,448 @@
+        unset existing_file_check_json_upload_file
+        # Check if file actually exists, and create if not.
+        existing_file_check_json_upload_file="$(_check_existing_file "${slug_upload_file}" "${folder_id_upload_file}" "${ACCESS_TOKEN}")"
+        if [ -n "${existing_file_check_json_upload_file}" ]; then
+            if [ -n "${SKIP_DUPLICATES}" ]; then
+                # Stop upload if already exists ( -d/--skip-duplicates )
+                _collect_file_info "${existing_file_check_json_upload_file}" "${slug_upload_file}"
+                "${QUIET:-_print_center}" "justify" "${slug_upload_file} already exists." "=" && return 0
+            else
+                request_method_upload_file="PATCH"
+                _file_id_upload_file="$(printf "%s\n" "${existing_file_check_json_upload_file}" | _json_value id 1 1)"
+                url_upload_file="${API_URL}/upload/drive/${API_VERSION}/files/${_file_id_upload_file}?uploadType=resumable&supportsAllDrives=true"
+                # JSON post data to specify the file name and folder under while the file to be updated
+                postdata_upload_file="{\"mimeType\": \"${mime_type_upload_file}\",\"name\": \"${slug_upload_file}\",\"addParents\": [\"${folder_id_upload_file}\"]}"
+                STRING="Updated"
+            fi
+        else
+            job_upload_file="create"
+        fi
+    fi
+
+    # Set proper variables for creating files
+    if [ "${job_upload_file}" = create ]; then
+        url_upload_file="${API_URL}/upload/drive/${API_VERSION}/files?uploadType=resumable&supportsAllDrives=true"
+        request_method_upload_file="POST"
+        # JSON post data to specify the file name and folder under while the file to be created
+        postdata_upload_file="{\"mimeType\": \"${mime_type_upload_file}\",\"name\": \"${slug_upload_file}\",\"parents\": [\"${folder_id_upload_file}\"]}"
+        STRING="Uploaded"
+    fi
+
+    _print_center "justify" "${slug_upload_file}" " | ${readable_size_upload_file}" "="
+
+    __file_upload_file="${HOME}/.google-drive-upload/${slug_upload_file}__::__${folder_id_upload_file}__::__${inputsize_upload_file}"
+    # https://developers.google.com/drive/api/v3/manage-uploads
+    if [ -r "${__file_upload_file}" ]; then
+        uploadlink_upload_file="$(cat "${__file_upload_file}" || :)"
+        http_code_upload_file="$(curl --compressed -s -X PUT "${uploadlink_upload_file}" --write-out %"{http_code_upload_file}")" || :
+        if [ "${http_code_upload_file}" = "308" ]; then # Active Resumable URI give 308 status
+            uploaded_range_upload_file="$(raw_upload_file="$(curl --compressed -s \
+                -X PUT \
+                -H "Content-Range: bytes */${inputsize_upload_file}" \
+                --url "${uploadlink_upload_file}" \
+                --globoff \
+                -D - || :)" && printf "%s\n" "${raw_upload_file}" | grep -io "range: bytes.*" | sed -e "s/.*ange: bytes=0-//" -e 's/'$'\r''//g')"
```

Comment:
Codacy found an issue: [In POSIX sh, $'..' is undefined.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)